astropy/config/paths.py

```diff
@@ -20,7 +30,9 @@
 ]
 
 
-def _get_dir_path(rootname: str, cls: type, fallback: str) -> Path:
+def _get_dir_path(
+    rootname: str, cls: type[_SetTempPath], fallback: Literal["cache", "config"]
```

Comment:
I think your new annotations are revealing a small design issue with these functions:
if `fallback` can only be exactly `'cache'` or `'config'`, shouldn't this mean that `cls` can only be exactly `set_temp_cache` or `set_temp_config` ? We could also naturally exclude invalid combinations like `cls=set_temp_cache, fallback='config'` by simply removing the `fallback` argument and making it a class attribute instead (in which case we can also keep the generalized annotation you used for `cls`)

Reply:
It may indeed be the case that the code can be refactored, but I'd prefer to keep the typing updates separate from code changes.

Reply:
Okay then, I'll merge this now and propose the refactor myself. Thanks !

Reply:
for reference this is https://github.com/astropy/astropy/pull/17554